control-plane/connect-inject/redirect_traffic.go

```diff
@@ -0,0 +1,120 @@
+package connectinject
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"strconv"
+
+	"github.com/hashicorp/consul/sdk/iptables"
+	corev1 "k8s.io/api/core/v1"
+)
+
+// addRedirectTrafficConfigAnnotation creates an iptables.Config based on proxy configuration.
+// iptables.Config:
+//   ConsulDNSIP: an environment variable named RESOURCE_PREFIX_DNS_SERVICE_HOST where RESOURCE_PREFIX is the consul.fullname in helm.
+//   ProxyUserID: a constant set in Annotations
+//   ProxyInboundPort: the service port or bind port
+//   ProxyOutboundPort: default transparent proxy outbound port or transparent proxy outbound listener port
+//   ExcludeInboundPorts: prometheus, envoy stats, expose paths, checks and excluded pod annotations
+//   ExcludeOutboundPorts: pod annotations
+//   ExcludeOutboundCIDRs: pod annotations
+//   ExcludeUIDs: pod annotations
+func (w *MeshWebhook) addRedirectTrafficConfigAnnotation(pod *corev1.Pod, ns corev1.Namespace) error {
+	cfg := iptables.Config{
+		ProxyUserID: strconv.Itoa(envoyUserAndGroupID),
+	}
+
+	// Set the proxy's inbound port.
+	cfg.ProxyInboundPort = proxyDefaultInboundPort
+
+	// Set the proxy's outbound port.
+	cfg.ProxyOutboundPort = iptables.DefaultTProxyOutboundPort
+
+	// TODO: update this
```

Comment:
This was your comment but I am not too sure if more needs to be added to the comment?

Reply:
I think I copy-pasta that comment from other place where we needed to do metrics so that's why I added a TODO to update it so it's relevant to this code